### ISSUE TITLE
regression when covariate is character showing object 'mean.cov' not found

### ISSUE THREAD

**Issue by biobibibi** [audrey-b/BUGSnet#60]
Hi all
It seems there is a bug when regression covariate is character or factor. I made sure there are less than 3 levels of factors but still couldn't run nma.model since  

"Error in structure(list(bugs = model, data = bugsdata2, scale = scale,  :  object 'mean.cov' not found"

Thank you!

**Comment by cqneosun** [comment]
have you figure the problem:can't load libraray(synapser)? I have the same probelm with you.Help!

**Comment by audrey-b** [comment]
Currently BUGSnet cannot handle factor covariates; only continuous.
See original article: https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/s12874-019-0829-2
I will flag this as a future improvement.

**Comment by Harmohit-Singh** [comment]
Implemented an informative error message.

**Comment by akshi96** [comment]
Has this been addressed? Just looking at the documentation and it says: Currently only categorical variables with fewer than 3 levels are supported. However, when trying to run this with a categorical covariate I keep getting the same error.

**Comment by Harmohit-Singh** [comment]
> Has this been addressed? Just looking at the documentation and it says: Currently only categorical variables with fewer than 3 levels are supported. However, when trying to run this with a categorical covariate I keep getting the same error.

Are you trying to run it with 2 categorical variables? Also, can you please confirm that you are using the latest version of BUGSnet?

**Comment by akshi96** [comment]
@Harmohit-Singh
Yep, currently trying to run it with a categorical variable with 2 levels. 
Version: 1.0.4

**Comment by akshi96** [comment]
> Are you trying to run it with 2 categorical variables? Also, can you please confirm that you are using the latest version of BUGSnet?

@Harmohit-Singh
Yep, currently trying to run it with a categorical variable with 2 levels. 
Version: 1.0.4

**Comment by audrey-b** [comment]
The reason 2 levels are supported (and not more) is because continuous variables are supported. So make sure your 2 level variable is coded as 0/1 and that you pass it as continuous (not factor). That should do it.

@Harmohit-Singh Let's keep this issue open and see if we can improve the code to allow 2-level factors with 0/1 to be passed to the function.

**Comment by akshi96** [comment]
@audrey-b  Thanks for the help. Changing it to a numeric type seems to have fixed the issue
